controller/src/beerocks/master/db/db.h

```diff
@@ -748,6 +758,8 @@ class db {
     // certification
     std::shared_ptr<uint8_t> certification_tx_buffer;
     std::unordered_map<sMacAddr, std::list<bss_info_conf_t>> bss_infos; // key=al_mac
+
+    master_thread *m_master_thread_ctx;
```

Comment:
Instead of making this a pointer, I'd make it a reference and pass it in the constructor (obviously) instead of set_master_thread_ctx(). This has the following advantages:

- Can remove the set function;
- It's guaranteed not to change over the lifetime;
- It indicates that the lifetime of db is tied to the lifetime of masther_thread;
- get_master_thread_ctx() returns a reference, so no need to check nullptr.

Reply:
It can't be done since the database is being created before the son_master is created and being passed as a reference to it.
I guess there is a solution for that, like not creating the database outside to the son_master thread and initialize it in the son_master constructor.
Do you want me to do it? It will need an extra commit on top.
@arnout @tomereli

Reply:
Sounds legit - the son master is the "owner" of the DB so it makes sense that it will create it in its constructor, so I would say go for it.

Reply:
I'm OK either way. It was anyway just a nitpick comment.